docs/guide/mocking.md

```diff
@@ -622,18 +627,40 @@ We can re-create this class with ES5 functions:
 ```ts
 const Dog = vi.fn(function (name) {
   this.name = name
+  // mock bound methods in the constructor, each instance will have its own spy
+  this.greet = vi.fn(() => `Hi! My name is ${name}!`)
 })
 
 // notice that static methods are mocked directly on the function,
 // not on the instance of the class
 Dog.getType = vi.fn(() => 'mocked animal')
 
 // mock the "speak" and "feed" methods on every instance of a class
-// all `new Dog()` instances will inherit these spies
+// all `new Dog()` instances will inherit and share these spies
 Dog.prototype.speak = vi.fn(() => 'loud bark!')
 Dog.prototype.feed = vi.fn()
 ```
 
+::: warning
+If a non-primitive is returned from the constructor function, that value will become the result of the new expression. In this case the `[[Prototype]]` may not be correctly bound:
```

Comment:
@sheremet-va 
I just copied and pasted this note from the [MDN docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Inheritance_and_the_prototype_chain#constructors), let me know if it's helpful to also add a link to this docs